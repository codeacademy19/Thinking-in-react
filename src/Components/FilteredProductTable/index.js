import React,{Component} from 'react';
import SearchBar from '../SearchBar';
import ProductTable from '../ProductTable';
class FilterableProductTable extends Component {
    // constructor(props) {
    //   super(props);
    //   this.state = {
    //     filterText: '',
    //     inStockOnly: false
    //   };
      
    //   this.handleFilterTextChange = this.handleFilterTextChange.bind(this);
    //   this.handleInStockChange = this.handleInStockChange.bind(this);
    // }
    state = {
            filterText: '',
            inStockOnly: false
    };
    handleFilterTextChange= (filterText) => {
      this.setState({
        filterText: filterText
      });
    }
    
    handleInStockChange=(inStockOnly) =>{
      this.setState({
        inStockOnly: inStockOnly
      })
    }
  
    render() {
      return (
        <div>
          <SearchBar
            filterText={this.state.filterText}
            inStockOnly={this.state.inStockOnly}
            handleFilterTextChange={this.handleFilterTextChange}
            handleInStockChange={this.handleInStockChange}
          />
          {this.state.filterText}
          <ProductTable
            product={this.props.product}
            filterText={this.state.filterText}
            inStockOnly={this.state.inStockOnly}
          />
        </div>
      );
    }
  }

  export default FilterableProductTable;