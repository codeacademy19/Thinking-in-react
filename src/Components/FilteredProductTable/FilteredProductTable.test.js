import React from 'react';
import renderer from 'react-test-renderer';
import FilteredProductTable from '.';
import {mount} from 'enzyme';
describe('ProductRow',()=>{
    const PRODUCTS = [
        {category: 'Sporting Goods', price: '$49.99', stocked: true, name: 'Football'},
        {category: 'Sporting Goods', price: '$9.99', stocked: true, name: 'Baseball'},
        {category: 'Sporting Goods', price: '$29.99', stocked: false, name: 'Basketball'},
        {category: 'Electronics', price: '$99.99', stocked: true, name: 'iPod Touch'},
        {category: 'Electronics', price: '$399.99', stocked: false, name: 'iPhone 5'},
        {category: 'Electronics', price: '$199.99', stocked: true, name: 'Nexus 7'}
      ];
    it('should render a tabel of items',()=>{
        const tree=renderer.create(<FilteredProductTable  product={PRODUCTS}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('should call "handleFilterTextChange" when the search text box is changed',()=>{
       const component= mount(<FilteredProductTable  product={PRODUCTS}/>) ;
       component.find(".TextArea").simulate('change',{target:{value:'ball'}});
       expect(component.instance().state.filterText).toEqual('ball');
       component.unmount();
    })
    it('should call "handleInStockChange" when the check box is changed',()=>{
        const component= mount(<FilteredProductTable  product={PRODUCTS}/>) ;
        component.find(".CheckBox").simulate('change',{target:{checked:true}});
        expect(component.instance().state.inStockOnly).toEqual(true);
        component.unmount();
     })


});