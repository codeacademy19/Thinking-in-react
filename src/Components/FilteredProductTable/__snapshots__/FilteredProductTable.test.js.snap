// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow should render a tabel of items 1`] = `
<div>
  <form>
    <input
      className="TextArea"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value=""
    />
    <p>
      <input
        checked={false}
        className="CheckBox"
        onChange={[Function]}
        type="checkbox"
      />
       
      Only show products in stock
    </p>
  </form>
  
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Price
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          colSpan="2"
        >
          Sporting Goods
        </th>
      </tr>
      <tr>
        <td>
          Football
        </td>
        <td>
          $49.99
        </td>
      </tr>
      <tr>
        <td>
          Baseball
        </td>
        <td>
          $9.99
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={
              Object {
                "color": "red",
              }
            }
          >
            Basketball
          </span>
        </td>
        <td>
          $29.99
        </td>
      </tr>
      <tr>
        <th
          colSpan="2"
        >
          Electronics
        </th>
      </tr>
      <tr>
        <td>
          iPod Touch
        </td>
        <td>
          $99.99
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={
              Object {
                "color": "red",
              }
            }
          >
            iPhone 5
          </span>
        </td>
        <td>
          $399.99
        </td>
      </tr>
      <tr>
        <td>
          Nexus 7
        </td>
        <td>
          $199.99
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
